services/gateway/shared/moleculer-web/methods.js

```diff
@@ -0,0 +1,121 @@
+
+/* eslint-disable */
+/*
+ * moleculer
+ * Copyright (c) 2020 MoleculerJS (https://github.com/moleculerjs/moleculer)
+ * MIT Licensed
+ */
+const _ = require('lodash');
+const kleur = require('kleur');
+const util = require('util');
+
+module.exports = {
+    methods: {
+        async httpHandler(req, res, next) {
+            req.$startTime = process.hrtime();
+            req.$service = this;
+            req.$next = next;
+            res.$service = this;
+            res.locals = res.locals || {};
+
+            let requestID = req.headers['x-request-id'];
+            if (req.headers['x-correlation-id']) requestID = req.headers['x-correlation-id'];
+            const options = { requestID };
+            if (this.settings.rootCallOptions) {
+                if (_.isPlainObject(this.settings.rootCallOptions)) {
+                    Object.assign(options, this.settings.rootCallOptions);
+                } else if (_.isFunction(this.settings.rootCallOptions)) {
+                    this.settings.rootCallOptions.call(this, options, req, res);
+                }
+            }
+            try {
+                const result = await this.actions.rest({ req, res }, options);
+                if (result == null) {
+                    if (this.serve) {
+                        this.serve(req, res, err => {
+                            this.logger.debug(err);
+                            this.send404(req, res);
+                        });
+                        return;
+                    }
+                    this.send404(req, res);
+                }
+            } catch (err) {
+                if (this.settings.log4XXResponses || (err && !_.inRange(err.code, 400, 500))) {
+                    this.logger.error(`<= ${this.coloringStatusCode(err.code)} Request error: ${err.name}: ${err.message} \n${err.stack} \nData: ${util.inspect(err.data)}`);
```

Comment:
Perhaps logging the request params/body would offer more assistance in investigating any future issues. @MichalTuleja Please share your thoughts.

```suggestion
                    this.logger.error(`<= ${this.coloringStatusCode(err.code)} Request error: ${err.name}: ${err.message} \n${err.stack} \nData: \nRequest params: ${util.inspect(req.$params)} \nRequest body: ${util.inspect(req.body)}`);
```